vmm/src/config.rs

```diff
@@ -2086,12 +2111,73 @@ impl RestoreConfig {
             .map_err(Error::ParseRestore)?
             .unwrap_or(Toggle(false))
             .0;
+        let net_ids = parser
+            .convert::<StringList>("net_ids")
+            .map_err(Error::ParseRestore)?
+            .map(|v| v.0);
+        let net_fds = parser
+            .convert::<IntegerList>("net_fds")
+            .map_err(Error::ParseRestore)?
+            .map(|v| v.0.iter().map(|e| *e as i32).collect());
 
         Ok(RestoreConfig {
             source_url,
             prefault,
+            net_ids,
+            net_fds,
         })
     }
+
+    pub fn validate(&self, vm_config: &VmConfig) -> ValidationResult<()> {
+        // Check if multiple same ids are passed
+        // verify if passed ids are valid
+        let vm_net_ids = match &vm_config.net {
+            Some(net_configs) => net_configs.iter().map(|v| v.id.as_ref().unwrap().clone()).collect(),
+            None => Vec::new(),
+        };
+        if let Some(net_ids) = &self.net_ids {
+            let mut net_ids_= BTreeSet::new();
+            for net_id in net_ids.iter() {
+                if !vm_net_ids.contains(net_id) {
+                    return Err(ValidationError::InvalidIdentifier(net_id.clone()));
+                }
+                if !net_ids_.insert(net_id.clone()) {
+                    return Err(ValidationError::IdentifierNotUnique(net_id.clone()));
+                }
+            }
```

Comment:
Also add a check that vm_net_ids.len() == net_ids.len(), to make sure all the net ids are passed while restoring the vm.

Reply:
Check to make sure all required net ids is at Line: 2162. (https://github.com/cloud-hypervisor/cloud-hypervisor/pull/6402/commits/b74a48d89e32e4da1514eff2622923f3de1b3867#diff-c235a65d9dfdc1392f1c9546df912679bbdd74138316a04d9cfd53ef8fe1f907R2162)
Also, restore does not except all the net ids. It expects net ids of the ones dealing with FDs.